### PR TITLE
Update Developer Guide

### LINKED ISSUE #163: Update on DG v3 Fazil

### PR DESCRIPTION
**Pull request by Md-Fazil**
1. Edit use cases in the developer's guide and remove unnecessary statements in existing use cases.
2. Add extensions to existing use cases.

Closes #163 .

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -309,95 +309,104 @@ otherwise)
...
+  Use case ends. 
 
-    * 3a1. StoreMando shows an error message.
-
-    * 3a2. StoreMando prompt the user for a correct input.
-
-      Use case resumes at step 3.
-
-* 3b. The command has an out-of-range index.
+* 3b. The index keyed in by the user does not exist in the displayed list.
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: should be 3a right?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -309,95 +309,104 @@ otherwise)
...
 
     * 3b1. StoreMando shows an error message.
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: same for this

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -309,95 +309,104 @@ otherwise)
...
-* 3a. The item details do not match any item in the storage.
+* 1a. The command keyed in by the user has an invalid syntax.
 
-    * 3a1. StoreMando shows an error message.
+    * 1a1. StoreMando shows an error message.
+    
+      Use case resumes at step 1.
 
-    * 3a2. StoreMando prompt the user for a correct input.
+* 1a. The new details keyed in by the user is the same as the existing details of the item.
```
Comment (kumsssss) [inline docs/DeveloperGuide.md]: it should be 1b for this i think

**Review by kumsssss — COMMENTED** [review]
minor tweaks needed. Otherwise, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Md-Fazil) [reply docs/DeveloperGuide.md]: Thanks and fixed.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Md-Fazil) [reply docs/DeveloperGuide.md]: Thanks and fixed.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Md-Fazil) [reply docs/DeveloperGuide.md]: Thanks and fixed.

**Review by JayChenYJ — COMMENTED** [review]
LGTM!

**Review by JayChenYJ — APPROVED** [review]

**Pull request merged**